pilot/pkg/config/kube/ingress/controller.go

```diff
@@ -112,6 +115,26 @@ func NewController(client kube.Client, meshWatcher mesh.Holder,
 		controllers.WithReconciler(c.onEvent),
 		controllers.WithMaxAttempts(5))
 	c.ingress.AddEventHandler(controllers.ObjectHandler(c.queue.AddObject))
+
+	// When the service which is referred by ingress using port name has been changed,
+	// we should trigger to rebuild rds.
+	c.services.AddEventHandler(controllers.ObjectHandler(func(o controllers.Object) {
```

Comment:
Event handler should ONLY call c.queue.AddObject. See `pkg/kube/controllers/example_test.go`.